src/wallet/wallet2.cpp

```diff
@@ -8670,36 +8683,21 @@ uint64_t wallet2::import_key_images(const std::vector<std::pair<crypto::key_imag
 			spent += amount;
 		else
 			unspent += amount;
+
 		LOG_PRINT_L2("Transfer " << i << ": " << print_money(amount) << " (" << td.m_global_output_index << "): "
 								 << (td.m_spent ? "spent" : "unspent") << " (key image " << req.key_images[i] << ")");
 
-		if(i < daemon_resp.spent_status.size() && daemon_resp.spent_status[i] == COMMAND_RPC_IS_KEY_IMAGE_SPENT::SPENT_IN_BLOCKCHAIN)
+		if(daemon_resp.spent_status[i] == COMMAND_RPC_IS_KEY_IMAGE_SPENT::SPENT_IN_BLOCKCHAIN)
```

Comment:
Why do you removed `i < daemon_resp.spent_status.size()`. What is if the daemon request will not have as much elements as `signed_key_images`, than we create an invalid memory access.

Reply:
This is already tested for at https://github.com/ryo-currency/ryo-currency/blob/7a197b3232a38182508ac36346c1679e9bdb47a6/src/wallet/wallet2.cpp#L8651

Reply:
I can confirm an exception is thrown if sizes are different

Reply:
but it is only tested if `check_spent` is true.

https://github.com/ryo-currency/ryo-currency/blob/7a197b3232a38182508ac36346c1679e9bdb47a6/src/wallet/wallet2.cpp#L8643